future-apps/payroll/contracts/Payroll.sol

```diff
@@ -634,4 +668,48 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
 
         return xrt;
     }
+
+    /**
+     * @dev Loop over tokens to send requested amount to the employee
+     * @param _employeeId Employee's identifier
+     * @param _totalAmount Total amount to be transferred to the employee distributed through the setup tokens allocation.
+     * @param _reference String detailing payment reason.
+     * @return True if there was at least one token transfer.
+     */
+    function _transferTokensAmount(uint256 _employeeId, uint256 _totalAmount, string _reference) private returns (bool somethingPaid) {
+        Employee storage employee = employees[_employeeId];
+        for (uint256 i = 0; i < allowedTokensArray.length; i++) {
+            address token = allowedTokensArray[i];
+            if (employee.allocation[token] != uint256(0)) {
+                uint128 exchangeRate = _getExchangeRate(token);
+                require(exchangeRate > 0, ERROR_EXCHANGE_RATE_ZERO);
+                // Salary converted to token and applied allocation percentage
+                uint256 tokenAmount = _totalAmount.mul(exchangeRate).mul(employee.allocation[token]);
+                // Divide by 100 for the allocation and by ONE for the exchange rate
+                tokenAmount = tokenAmount / (100 * ONE);
+                finance.newPayment(token, employee.accountAddress, tokenAmount, 0, 0, 1, _reference);
+                emit SendPayment(employee.accountAddress, token, tokenAmount, _reference);
+                somethingPaid = true;
+            }
+        }
+    }
+
+    /**
+     * @dev Try removing employee if there are no pending payments and has reached employee's end date.
+     * @param _employeeId Employee's identifier
+     * @param _date Date timestamp used to evaluate if the employee can be removed from the payroll.
+     */
+    function _tryRemovingEmployee(uint256 _employeeId, uint64 _date) private {
+        Employee storage employee = employees[_employeeId];
+
+        bool hasReachedEndDate = employee.endDate <= _date;
+        bool isOwedSalary = _getOwedSalary(_employeeId, _date) > 0;
```

Comment:
Wouldn't it be more straightforward to do something like: 
```
bool noPendingSalary = _getOwedSalary(_employeeId, _date) == 0;
bool noPendingAccruedValue = employees[_employeeId].accruedValue == 0;

if (hasReachedEndDate && noPendingSalary && noPendingAccruedValue) {
(...)
```
?

Reply:
Sure 😅, I'll address this comment in this [polishing PR](https://github.com/aragon/aragon-apps/pull/744)

Reply:
Nice. Another thing I missed: you can use the pointer `employee` in a couple of places where you are using `employees[_employeeId]`.

Reply:
Yea! I already did that in #744 :)

Reply:
Oh, sorry, as it was "WIP", I didn't check it out yet.